docs/developer-guide/etendo-classic/concepts/data-model.md

```diff
@@ -0,0 +1,513 @@
+
+#### Defining a Record Identifier
+
+Record Identifiers are defined in `Application Dictionary` > `Tables and Columns` > `Table` > `Column`. To set a column as part of the record identifier for its table mark the **Identifier** check.
+
+As many columns in the same table can be part of the record identifier, the way to define the order they will be displayed in it is set in the **Sequence number** field.
+
+### Callout
+
+A Callout is a piece of code that is triggered when a field is modified. The callout is associated to the column which associated field will trigger the action.
+
+##### What Callouts can do
+
+Callouts are implemented by Java classes, so they can do virtually anything. But it must be taken into account that they are triggered when the field for the column they are associated with is modified. This happens before saving it to database, so it is possible the record in database does not already exist. Taking this into account, they should query in database for the current record.
+
+They are typically used to:
+
+- Load or change dynamically information in UI after a user change. For example, set a default price when a product is selected.
+- Show messages after modifying a field.
+- Execute a Javascript function.
+- Move the cursor to a field.
+
+###### Callouts vs Triggers
+
+Both callouts and triggers are executed when there are changes in data. The main difference between callouts and database triggers is that callouts are executed in the front end whereas triggers are in the back end. This means that callouts should affect primary to data displayed in UI and triggers to data in database, additionally callouts are only executed when the user is modifying data manually through UI but they are not exectued in case the same data is modified by any other process.
+
+!!!important
+    Due to all these reasons, data integrity should not be relied only on callouts. It should be also implemented in database, through triggers or constraints.
+
+##### Defining a Callout
+
+There is a [guide](../how-to-guides/How_to_create_a_Callout.md) related to callouts you can read to have an step by step example for a callout creation.
+
+###### Application Dictionary
+
+Callouts are defined in `Application Dictionary` > `Setup` > `Callout` window and then associated with the column that will invoke them when it is modified.
+
+###### Java Implementation
+
+Callouts are implemented by a Java Servlet.
+
+!!!note
+    The simplest way to implement a callout is using the **SimpleCallout** class, that keeps you out from the 'plumbing' of building the response.
+
+The things to consider are:
+
+- The **command** callouts are invoked with is always **DEFAULT**.
+- Callouts receive a parameter called **inpLastFieldChanged**, this parameter has the name of the field that triggered the callout. Using this parameter, it is possible to reuse the same callout for different columns, this makes sense in case the action for both columns is related.
```

Comment:
```suggestion
- Callouts receive a parameter called `inpLastFieldChanged`, this parameter has the name of the field that triggered the callout. Using this parameter, it is possible to reuse the same callout for different columns, this makes sense in case the action for both columns is related.
```